File: examples/jte-example/src/main/java/de/tschuehly/example/jte/web/simple/SimpleViewComponent.java
package de.tschuehly.example.jte.web.simple;

import de.tschuehly.spring.viewcomponent.core.component.ViewComponent;
import de.tschuehly.example.jte.core.ExampleService;
import de.tschuehly.spring.viewcomponent.jte.ViewContext;

@ViewComponent
public class SimpleViewComponent {
    final ExampleService exampleService;

    public SimpleViewComponent(ExampleService exampleService) {
        this.exampleService = exampleService;
    }

    public record SimpleView(String helloWorld) implements ViewContext {}

    public SimpleView render() {
        return new SimpleView(exampleService.getHelloWorld());
    }
}

File: examples/jte-example/src/main/kotlin/de/tschuehly/kteviewcomponentexample/web/simple/SimpleViewComponent.kt
package de.tschuehly.kteviewcomponentexample.web.simple

import de.tschuehly.kteviewcomponentexample.core.ExampleService
import de.tschuehly.spring.viewcomponent.core.IViewContext
import de.tschuehly.spring.viewcomponent.core.action.PostViewAction
import de.tschuehly.spring.viewcomponent.core.component.ViewComponent
import de.tschuehly.spring.viewcomponent.jte.ViewContext

@ViewComponent
class SimpleViewComponent(
    val exampleService: ExampleService
) {
    fun render() = HelloWorldView(exampleService.getHelloWorld())

    data class HelloWorldView(val helloWorld: String) : ViewContext

    @PostViewAction
    fun testAction(): IViewContext {
        return render()
    }
}